### ISSUE TITLE
Privacy consideration about tracking location

### ISSUE THREAD

**Issue by jyasskin** [w3c/system-wake-lock#3]
If an app has the `"system"` lock or higher, it can send network requests to track the user's current IP address, and can probably use `geolocation.watchPosition()` to track the user's physical location directly. Users must consent to this when the wakelock is active.

**Comment by xfq** [comment]
Thanks. I have transferred the issue.

**Comment by EricMwobobia** [comment]
@jyasskin You raise a valid point - given the app requires the IP for network connection purposes, would limiting the data sent out by the geolocation.watchPosition(); success callback to only the features with customer consent to help preserve user privacy. Is this feasible?

**Comment by jyasskin** [comment]
I think there's a privacy worry about even giving out the IP address (e.g. https://whatismyipaddress.com/ probably shows your city), although it's clearly less than the geolocation API. I could imagine having the `system` wakelock end if the system's IP address changes, but that could break use cases, so an explicit dialog might still be useful.

**Comment by EricMwobobia** [comment]
@jyasskin True, let's explore further. It's interesting how we'll tackle the privacy concerns on the IP as the app server will require the IP in order to exchange information with the user. It's a delicate balance.

**Comment by hadriann** [comment]
All things aside, the spec for system wake lock seems stagnant. And with no spec, there will be no implementation. @anssiko any idea why is this happening and what is needed to get the things moving?

**Comment by anssiko** [comment]
@hadriann, I opened a separate issue to address your question, please see https://github.com/w3c/system-wake-lock/issues/4 and follow up there as appropriate.
